source/docs/software/basic-programming/java-gc.rst

```diff
@@ -0,0 +1,35 @@
+Java Garbage Collection
+=======================
+Java garbage collection is the process of automatically managing memory for Java objects. The Java Virtual Machine (JVM) is responsible for creating and destroying objects, and the garbage collector is responsible for identifying and reclaiming unused objects.
+
+Java garbage collection is an automatic process, which means that the programmer does not need to explicitly deallocate memory. The garbage collector keeps track of which objects are in use and which are not, and it periodically reclaims unused objects.
+
+Object Creation
+---------------
+
+Creating a large number of small objects in Java can lead to memory and performance issues. While the Java Garbage Collector (GC) is designed to handle memory management efficiently, creating too many small objects can overwhelm the GC and cause performance degradation.
+
+Memory Concerns
+~~~~~~~~~~~~~~~
+
+When a large number of small objects are created, it increases the overall memory footprint of the application. This is because each object has its own overhead. While the overhead for a single object may be insignificant, it can become substantial when multiplied by a large number of objects.
+
+The GC's job is to periodically identify and reclaim unused objects in memory. However, when the GC has to collect a large number of small objects frequently, it can become a performance bottleneck. This is because the GC has to perform more work to collect and process each object, which can slow down overall application performance.
+
+Performance Concerns
+~~~~~~~~~~~~~~~~~~~~
+
+The performance impact of creating a lot of small objects is not limited to GC overhead. Creating and destroying objects also involves CPU cycles, which can further impact application performance. Additionally, accessing and manipulating large numbers of small objects can lead to cache misses, which can further slow down the application.
+
+Design Considerations
+~~~~~~~~~~~~~~~~~~~~~
+
+If you anticipate your application creating a large number of small objects, it is important to consider design strategies to mitigate the potential memory and performance issues. Here are some strategies to consider:
+
+- Minimize object creation: Carefully evaluate the need for each object creation. If possible, reuse existing objects or use alternative data structures, such as arrays or primitives, to avoid creating new objects.
+
+- Object pooling: Implement object pooling to reuse frequently created objects. This involves creating a pool of objects that are available for use when needed. When an object is no longer needed, it is returned to the pool instead of being destroyed.
```

Comment:
While certainly a general design pattern, I'm struggling to think of a use case for an FRC robot. I'm thinking this is probably not something who'd reading about the GC for the first time would be prepared to implement, so I'd lean towards not covering in this artcile